src/test/scala/com/fulcrumgenomics/bam/SamRecordClipperTest.scala

```diff
@@ -588,24 +588,74 @@ class SamRecordClipperTest extends UnitSpec with OptionValues {
     mate.cigar.toString shouldBe "1S99M"
   }
 
+  it should "clip reads that overlap with soft-clipping on the forward read after the midpoint" in {
+    val (rec, mate) = pair(1, "95M5S", Plus, 50, "100M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (23, 23)
+    rec.start shouldBe 1
+    rec.end shouldBe 72
+    rec.cigar.toString shouldBe "72M28H"
+    mate.start shouldBe 73
+    mate.cigar.toString shouldBe "23H77M"
+  }
+
+  it should "clip reads that overlap with soft-clipping on the reverse read before the midpoint" in {
+    val (rec, mate) = pair(1, "100M", Plus, 55, "5S95M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (23, 23)
+    rec.start shouldBe 1
+    rec.end shouldBe 77
+    rec.cigar.toString shouldBe "77M23H"
+    mate.start shouldBe 78
+    mate.cigar.toString shouldBe "28H72M"
+  }
+
+  it should "clip reads that overlap 1 bp directly in the middle of the pair" in {
+    val (rec, mate) = pair(1, "99M1S", Plus, 99, "1S99M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (0, 1)
+    rec.start shouldBe 1
+    rec.end shouldBe 99
+    rec.cigar.toString shouldBe "99M1H"
+    mate.start shouldBe 100
+    mate.cigar.toString shouldBe "2H98M"
+  }
+
+  it should "clip reads that overlap in the first half of the pair, not over the middle" in {
```

Comment:
What does "overlap in the first half of the pair," mean?

Reply:
Those ascii images may be very helpful as comments in the tests

Reply:
I think for these tests it would be really helpful to add a comment inside the test something like this:

```scala
// Template spans chr1:-169, with mid-point=85
// Second read doesn't overlap the mid-point so all clipping should be done to the first read
```

Though having written that, I'm now curious why the mate/R2 gets clipped at all in this test?

Reply:
Off-by-1 error.  It wasn't technically incorrect since it would do the same amount of trimming, but I've corrected it now so that it only trims the longer read if possible.